providers/implementations/macs/kmac_prov.c

```diff
@@ -286,6 +286,17 @@ static int kmac_init(void *vmacctx, const unsigned char *key,
     block_len = EVP_MD_block_size(ossl_prov_digest_md(&kctx->digest));
     if (block_len < 0)
         return 0;
+    /*
+     * The temporary buffer for the customisation and header needs to contain:
+     *      len(encoded_int(block_len)) + len(encoded_string(custom))
+     *      + len(kmac_encoded_string)
+     * bytes rounded up to the next block_size multiple.
+     */
+    out_max = ((2 + kctx->custom_len + sizeof(kmac_string) + block_len - 1)
```

Comment:
Normally I would make the bytepad function figure out this number if out is NULL..

Reply:
Can't say I like this this approach, but it is used elsewhere....

Reply:
And done.